src/Uno.UI/UI/Xaml/Controls/TextBox/TextBox.macOS.cs

```diff
@@ -1,56 +1,192 @@
 		}
 
-		partial void OnForegroundColorChangedPartial(Brush newValue)
+		partial void OnTextAlignmentChangedPartial(DependencyPropertyChangedEventArgs e)
+		{
+		}
+
+		private void UpdateTextBoxView()
+		{
+			if (_contentElement != null)
+			{
+				if (_textBoxView is TextBoxView)
+				{
+					return;
+				}
+
+				_textBoxView = new TextBoxView(this) { UsesSingleLineMode = AcceptsReturn || TextWrapping != TextWrapping.NoWrap };
+
+				_contentElement.Content = _textBoxView;
+				_textBoxView.SetTextNative(Text);
+				InitializeProperties();
+			}
+		}
+
+		internal bool OnKey(char key)
 		{
+			var keyRoutedEventArgs = new KeyRoutedEventArgs(this, key.ToVirtualKey())
+			{
+				CanBubbleNatively = true
+			};
 
+			var downHandled = RaiseEvent(KeyDownEvent, keyRoutedEventArgs);
+
+			keyRoutedEventArgs.Handled = false; // reset to unhandled for Up
+			var upHandled = RaiseEvent(KeyUpEvent, keyRoutedEventArgs);
+
+			return downHandled || upHandled;
 		}
 
 		partial void UpdateFontPartial()
 		{
+			if (_textBoxView != null)
+			{
+				_textBoxView.UpdateFont();
+			}
+		}
 
+		partial void OnMaxLengthChangedPartial(DependencyPropertyChangedEventArgs e)
+		{
+			//support by MultilineTextBoxDelegate and SinglelineTextBoxDelegate
 		}
 
-		partial void OnTextWrappingChangedPartial(DependencyPropertyChangedEventArgs e)
+		partial void OnIsReadonlyChangedPartial(DependencyPropertyChangedEventArgs e)
 		{
+			//support by MultilineTextBoxDelegate and SinglelineTextBoxDelegate
+		}
 
+		partial void OnIsTextPredictionEnabledChangedPartial(DependencyPropertyChangedEventArgs e)
+		{
+			// There doesn't seem to be any way to disable/enable TextPrediction without disabling/enabling SpellCheck
+			if (!IsTextPredictionEnabledErrorMessageShown)
+			{
+				this.Log().Warn("IsTextPredictionEnabled isn't supported on macOS.");
+				IsTextPredictionEnabledErrorMessageShown = true;
+			}
 		}
 
-		partial void OnTextAlignmentChangedPartial(DependencyPropertyChangedEventArgs e)
+		private static bool IsTextPredictionEnabledErrorMessageShown = false;
+
+		public int SelectionStart
 		{
+			get
+			{
+				if (_textBoxView?.SelectedRange == null)
+				{
+					return 0;
+				}
+
+				return (int)_textBoxView?.SelectedRange.Location;
+			}
+			set
+			{
+				if(_textBoxView is TextBoxView sltbv)
+				{
+					sltbv.SelectWithFrame(sltbv.Frame, sltbv.CurrentEditor, null, value, _textBoxView.SelectedRange.Length);
+				}
+			}
+		}
 
+		public int SelectionLength
+		{
+			get
+			{
+				if (_textBoxView?.SelectedRange == null)
+				{
+					return 0;
+				}
+
+				return (int)_textBoxView?.SelectedRange.Location;
+			}
+			set
+			{
+				if (_textBoxView is TextBoxView sltbv)
+				{
+					sltbv.SelectWithFrame(sltbv.Frame, sltbv.CurrentEditor, null, _textBoxView.SelectedRange.Location, value);
+				}
+			}
 		}
 
-		partial void OnIsSpellCheckEnabledChangedPartial(DependencyPropertyChangedEventArgs e)
+		//protected void SetSecureTextEntry(bool isSecure)
+		//{
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Remove this commented out code.](https://app.codacy.com/gh/unoplatform/uno/pullRequest?prid=5303491)